### PR TITLE
Update rat-facts:

### PR DESCRIPTION
**Pull request by dewiniaid**
- Load and validate ratfacts at setup() time rather than on every fact request
- Overhaul reciteFact to use commands decorator instead of rule.
- When listing facts, sort them.

### CODE REVIEW

File: sopel-modules/rat-facts.py [hunk 1]
```diff
@@ -10,53 +10,58 @@
...
-        with open(bot.config.ratfacts.filename) as f:
-            facts = json.load(f)
-    except IOError:
-        # We couldn't open facts.json...
-        return bot.reply('There appears to be a problem with the fact list.')
-    return bot.reply('Known facts: '+', '.join(facts.keys()))
-
+    facts = bot.memory['ratbot']['facts']
+    if not facts:
+        return bot.reply("Like Jon Snow, I know nothing.  (Or there's a problem with the fact list.")
```
Comment (tyrope) [inline sopel-modules/rat-facts.py]: Missing closing paren.

**Comment by tyrope** [comment]
This will conflict with #45 and remove the 'accidental feature' of being able to update the facts list without needing a module reload.

**Comment by dewiniaid** [comment]
The new commits add !fact reload (if you have halfop or better on any channel) and -- if the filename is a directory, scan all files in that directory.  This should supercede the need for #45 while not requiring immediate configuration changes.

**Comment by dewiniaid** [comment]
Closing this in favor of #49

**Pull request closed without merging**